Scripts/Isle of Quel'Danas: Thalorien Dawnseeker

Sorikoff opened the pull request:
**Changes proposed:** An attempt to implement quest Thalorien Dawnseeker. All events are linked to Thalorien Dawnseekers Remains which are also correctly positioned right now. 

**Target branch(es):** 3.3.5 (can be ported though)

**Tests performed:** Does work.

**Issues addressed:** Closes #12444

**Known issues and TODO list:** (add/remove lines as needed)

- [x] Sunwell Defenders right now have not exactly correct paths. Looks to me each of them must have own waypoints (although they run away anyway, huh...).

jackpoz committed:
Core/Misc: Headers cleanup

Use simple std::string concatenation instead of stringstream to avoid including <sstream> header.

Ovahlord committed:
Core/Groups: do not allow lfg groups to change the loot mode (#23336)

While the UI does not allow changing the loot mode you can still trigger the opcode by running interface scripts which we will now block as well so nobody can ninja-loot anymore.

jackpoz committed:
Scripts/HallsOfLightning: Fix Volkhan spamming in chat when wiping at 10% health

Ref #23082

jackpoz committed:
Core/Map: Don't log errors about Area ID 0

Restore a check about Area ID 0 that would silently ignore this case instead of spamming log errors. Check removed in e79c595b695ea5f35b8324efbb7741ccb102bce9
Fix #23085

Killyana committed:
DB/Quest: The Black Box

The quest is not repeatable

ZenoX92 committed:
DB/Quest: A Mission of Mercy

Closes #23329

illfated committed:
DB/Quest: Old Whitebark's Pendant (8474) - spacing & words missing in RewardText

Closes #23339

Shauren committed:
Build: Fixed VS always triggering a full build when PCH is enabled, even if only a source file was edited

Wyrserth committed:
Core/SAI: add action_param3 to SMART_ACTION_REMOVEAURASFROMSPELL to allow it to remove only auras from spells that were cast by the entity itself. (#23345)

Wyrserth committed:
Script/Creature: calculate Traveler's Tundra Mammoth's NPCs' exit position based on current player's position (#23344)

* Script/Creature: calculate Traveler's Tundra Mammoth's NPCs' exit position based on current player's position.

* Comment out unused argument.

Shauren committed:
Build: Fixed dynamic PCH builds

Sorikoff committed:
Initial work

Killyana committed:
DB/Quests: Remove deprecated quests from Horde Warbringer

Sorikoff committed:
Kill Credit

Killyana committed:
DB/Creature: Bash'ir Surveyor

Wyrserth committed:
Script/Creature: implement spell effect for the item Prototype Neural Needler used on Imprisoned Beryl Sorcerer (#23349)

* Script/Creature: implement spell effect for the item Prototype Neural Needler used on Imprisoned Beryl Sorcerer.

* Rename 9999_99_99_99_world.sql to 2019_06_03_02_world.sql

jackpoz commented on src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp:
```diff
@@ -0,0 +1,367 @@
+                    case EVENT_KNEEL:
+                        if (_thalorien)
+                            _thalorien->SetStandState(UNIT_STAND_STATE_KNEEL);
+                            _thalorien->DespawnOrUnsummon(5 * IN_MILLISECONDS);
```
I think you are missing some brackets { } here

jackpoz commented on src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp:
```diff
@@ -0,0 +1,367 @@
+                    case EVENT_OUTRO_3:
+                        if (_thalorien)
+                            _thalorien->AI()->Talk(SAY_THALORIEN_9);
+                            _thalorien->AI()->DoCastSelf(SPELL_POLYMORPH_VISUAL);
```
I think you are missing some brackets { } here

jackpoz commented on src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp:
```diff
@@ -0,0 +1,367 @@
+                    case EVENT_MORLEN_ATTACK:
+                        if (_morlen)
+                            _morlen->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_IMMUNE_TO_NPC);
+                            if (_thalorien)
```
I think you are missing some brackets { } here

jackpoz commented on src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp:
```diff
@@ -0,0 +1,367 @@
+                    case EVENT_SUMMON_MORLEN:
+                        if (Creature* morlen = me->SummonCreature(NPC_MORLEN_GOLDGRIP, morlenSummon, TEMPSUMMON_CORPSE_DESPAWN, 0))
+                            _morlen = morlen;
+                            _morlen->AI()->DoCastSelf(SPELL_BLOOD_PRESENCE);
```
I think you are missing some brackets { } here

Sorikoff committed:
Improvements

Sorikoff commented on src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp:
```diff
@@ -0,0 +1,367 @@
+        EventMap _events;
+        SummonList _summons;
+        ObjectGuid _playerGUID;
+        Creature* _thalorien;
```
Good point. Can you tell me whether replacing pointers to ObjectAccessor's could be a solution?

Sorikoff replied:
Yeah it's fine. I was afraid of performance, but profiler didn't confirm that.

Sorikoff committed:
Missing Header

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,111 @@
+DELETE FROM `spell_area` WHERE `spell`=70193 AND `area`=4092;
+INSERT INTO `spell_area` (`spell`,`area`,`quest_start`,`quest_end`,`aura_spell`,`racemask`,`gender`,`autocast`,`quest_start_status`,`quest_end_status`) VALUES
+(70193,4092,24535,0,0,0,2,1,10,0),
+(70193,4092,24563,0,0,0,2,1,10,0);
```
Does the phasing ends when the quest is rewarded? no nee to set `quest_end_status`=11?

Sorikoff replied:
This spell is also used in the next quest.

Wyrserth committed:
Script/GunDrak: fix order for yell and altar texts on death for two bosses (#23357)

Killyana committed:
DB/Creature: Or'kaos the Insane

Originally Toranaku was supposed to give a quest to kill Or'kaos upon a player reaching Exalted with Netherwing in order for him or her obtain a netherdrake mount; but the idea was replaced with : Lord Illidan Stormrage quest.
Currently serves no useful purpose, including lacks loot.

Killyana committed:
DB/Creature: Seer Nakha

Sorikoff committed:
Creature* -> ObjectGuid

Riztazz commented on src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp:
```diff
@@ -0,0 +1,374 @@
+
+        bool GossipSelect(Player* player, uint32 menuId, uint32 /*gossipListId*/) override
+        {
+            if (menuId == MENU_EXAMINE_REMAINS)
```
What if two players have gossip window open? and both click it

Sorikoff replied:
After first one clicks, menu is closed and event begins.

Riztazz replied:
Is it closed for the other player tho?

Sorikoff replied:
Closed for both meaning you won't be able to start it again shortly.

jackpoz commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -20,9 +20,9 @@ INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,
 (15,10908,0,0,1,9,0,24563,0,0,0,0,0,'','Show gossip option if 24563 is incomplete');
 
-DELETE FROM `gameobject` WHERE `id`=201718;
+DELETE FROM `gameobject` WHERE `guid`=9004;
```
please delete by guid AND id, just to make sure we are deleting exactly what we want and not some other gameobject

Sorikoff committed:
Improvements #2

Sorikoff committed:
DB

Killyana committed:
DB/Trainers: Set the correct greeting text for Enchanting Trainers

Killyana committed:
DB/Creature: Update Silvermane Stalker model with the correct one

Closes #23353

Killyana committed:
DB/Creature: Remove some wrong waypoint actions

Killyana committed:
Spell/Script: Stasis Field

Closes #23355

Wyrserth committed:
Script/Quest: implement event for the quest Orabus the Helmsman (#23346)

* Script/Quest: implement event for the quest Orabus the Helmsman.

Thanks Wyreth for the original script.

* Fix coordinates that I inverted by mistake.

* Rename 9999_99_99_99_world.sql to 2019_06_04_04_world.sql

Sorikoff committed:
Restore Gossip immediately

ForesterDev committed:
Core/Logging: Split logs for unknown and zero triggered spells (#23180)

* Core/Conditions: implement CONDITION_GAMEMASTER

* Split logs for unknown and zero triggered spells

* Update SpellAuraEffects.cpp

* Update SpellAuraEffects.cpp

* Update SpellAuraEffects.cpp

* Update SpellEffects.cpp

* Update SpellAuraEffects.cpp

* Update SpellAuraEffects.cpp

* Update SpellEffects.cpp

* Update SpellEffects.cpp

Killyana commented on src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp:
```diff
@@ -0,0 +1,410 @@
+
+        void Reset() override
+        {
+            me->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
```
Add the npc flag on the DB

Sorikoff replied:
It's already set, isn't it?
```sql
mysql> SELECT `npcflag` FROM `creature_template` WHERE `entry`=37552;
+---------+
| npcflag |
+---------+
|       1 |
+---------+
1 row in set (0.00 sec)
```

Sorikoff committed:
Imrovements #3

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,111 @@
+DELETE FROM `creature_template_addon` WHERE `entry`=37552;
+INSERT INTO `creature_template_addon` (`entry`,`bytes2`,`auras`) VALUES (37552,1,'29266');
+
+DELETE FROM `gossip_menu` WHERE `MenuID`=10908;
```
Remove the gossip 37552 as it's wrong
``` sql
DELETE FROM `gossip_menu` WHERE `MenuID` IN (10908,37552 );

Sorikoff committed:
Also remove wrong gossip

Wyrserth committed:
Script/ZulGurub: allow Hakkar to cast Mind Control only if more than one unit is currently fighting him.

Also fix target type - it should always target the current tank.

Killyana committed:
DB/Misc: Arcatraz entrance event

Closes#21107

Wyrserth committed:
DB/Quest: A Lesson in Fear

Closes #23366

Sorikoff committed:
DB/Quest: Apothecary Zelana

Closes #23361

Killyana committed:
DB/QuestGreeting: More Quest Greeting

Closes #23340

ccrs committed:
Scripts/ZulGurub: general instance related refactoring/updating

plus couple event-spells-ai related corrections
plus codestyle cleanup

ccrs committed:
Core/Log: 1f85348 followup

add missing logger entries

Wyrserth committed:
Script/ZulGurub: implement several RP texts (#23358)

ccrs committed:
Scripts/ZulGurub: 41931c0 followup

Wyrserth committed:
Script/Dalaran: don't allow Dalaran faction guards to teleport a player that has the Trespasser! debuff. (#23373)

jackpoz committed:
Core/Logs: Allow to log startup logs into Database appender

Wyrserth committed:
DB/Item: Add gold loot to item Chest of Spoils

Closes #23369

Wyrserth committed:
DB/SAI: out of combat dialog for NPCs Crusade Engineer Spitzpatrick and Crusade Architect Silas

Closes #23370

Wyrserth committed:
DB/Gossip: add Death Knight-only gossip text for NPC The Ebon Watcher

Closes #23371

Wyrserth committed:
Loot/Item: fix drop chance for Moonrest Gardens Plans

Closes #23372

Sorikoff committed:
DB/SAI: Nerub'ar

Closes #23374

Sorikoff committed:
DB/SAI: Howling Fjord

Closes #23376

Sorikoff committed:
Fix UB

Sorikoff commented:
_An accident happened._ Sorry.

Superseded by #23382.
